zeppelin/executors/notebook_executor.py

```diff
@@ -0,0 +1,85 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+
+import os
+import requests
+import sys
+import json
+import time
+
+
+class NotebookExecutor():
+    """NotebookExecutor is a command line tool to execute a Zeppelin notebook."""
+
+    def __init__(self, notebook_id, output_path, zeppelin_url):
+        """Initialize class object with attributes based on CLI inputs."""
+        self.notebook_id = notebook_id
+        self.output_path = output_path
+        self.zeppelin_url = zeppelin_url
+
+    def run_notebook(self):
+        """Call API to execute notebook."""
+        requests.post('http://{0}/api/notebook/job/{1}'.format(
+                      self.zeppelin_url, self.notebook_id))
+
+    def wait_for_notebook_to_execute(self):
+        """Wait for notebook to finish executing before continuing."""
+        while True:
+            r = requests.get('http://{0}/api/notebook/job/{1}'.format(
+                             self.zeppelin_url, self.notebook_id))
+            data = r.json()['body']
+            if all(paragraph['status'] in ['FINISHED', 'ERROR'] for paragraph in data):
+                break
+            time.sleep(1)
+
+    def get_executed_notebook(self):
+        """Return the executed notebook."""
+        r = requests.get('http://{0}/api/notebook/{1}'.format(
+                         self.zeppelin_url, self.notebook_id))
+        if r.status_code == 200:
+            return r.json()['body']
+        else:
+            print('ERROR: Could not get executed notebook.', file=sys.stderr)
+            sys.exit(1)
+
+    def save_notebook(self, body):
+        """Save notebook depending on user provided output path."""
+        try:
+            directory = os.path.dirname(self.output_path)
+            full_path = os.path.join(directory, 'note.json')
+
+            with open(full_path, 'w') as fh:
+                fh.write(json.dumps(body, indent=2))
+        except:
```

Comment:
I'm not a big fan of catchall try/except blocks as they tend to hide bugs. Could you please be more specific with the exceptions you wanted to catch?